src/Plugin/Action/SendEmail.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+/**
+ * @file
+ * Contains Drupal\rules\Plugin\Action\SendEmail.
+ */
+
+namespace Drupal\rules\Plugin\Action;
+
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\MailManager;
+use Drupal\Core\MailManagerInterface;
+use Drupal\rules\Core\RulesActionBase;
+use Psr\Log\LoggerInterface;
+use Psr\Log\LogLevel;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+use Drupal\Core\Render\Element;
+
+/**
+ * Provides "Send email" rules action.
+ *
+ * @Action(
+ *   id = "rules_send_email",
+ *   label = @Translation("Send email"),
+ *   context = {
+ *     "send_to" = @ContextDefinition("email",
+ *       label = @Translation("Send to"),
+ *       description = @Translation("Email address(es) drupal will send an email to."),
+ *       multiple = TRUE,
+ *     ),
+ *     "subject" = @ContextDefinition("string",
+ *       label = @Translation("Subject"),
+ *       description = @Translation("The email's subject."),
+ *     ),
+ *     "message" = @ContextDefinition("string",
+ *       label = @Translation("Message"),
+ *       description = @Translation("The email's message body."),
+ *     ),
+ *     "reply" = @ContextDefinition("email",
+ *       label = @Translation("Reply to"),
+ *       description = @Translation("The mail's reply-to address. Leave it empty to use the site-wide configured address."),
+ *       required = FALSE,
+ *     ),
+ *     "language" = @ContextDefinition("language",
+ *       label = @Translation("Language"),
+ *       description = @Translation("If specified, the language used for getting the mail message and subject."),
+ *       required = FALSE,
+ *     ),
+ *   }
+ * )
+ *
+ * @todo: Define that message Context should be textarea comparing with textfield Subject
+ * @todo: Add access callback information from Drupal 7.
+ * @todo: Add group information from Drupal 7.
+ */
+class SendEmail extends RulesActionBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * @var LoggerInterface $logger
```

Comment:
This should use the fully qualified namespace, and the variable name should be removed. Also a description is missing. See https://www.drupal.org/coding-standards/docs#var